cmd/maya-exporter/app/command/command.go

```diff
@@ -118,10 +128,34 @@ func Run(cmd *cobra.Command, options *VolumeExporterOptions) error {
 func (o *VolumeExporterOptions) RegisterJivaStatsExporter() error {
 	controllerURL, err := url.ParseRequestURI(o.ControllerAddress)
 	if err != nil {
-		log.Println(err)
+		glog.Info(err)
 		return errors.New("Error in parsing the URI")
 	}
 	exporter := collector.NewExporter(controllerURL)
 	prometheus.MustRegister(exporter)
 	return nil
 }
+
+// RegisterCstorStatsExporter creates a connection with the unix domain socket
+// on the socketPath and the register the exporter for the collection of metrics.
+// Fetching the large size images from the dockerhub takes time, so this retries
+// to connect if the istgt target has not started yet.
```

Comment:
Is there any policy in kubernetes to launch & start containers in order.
In this case the istgt container needs to be up & running before launching the exporter container.
It will be great if exporter & istgt dependency can be managed in a better way.

Reply:
yes @AmitKumarDas Initcontainers are useful for such scenarios i have used it a bit in EFK stack setup. This scenario is something that can only be verified if we have full fledged cstor available. I have tested this on local-setup. Primary idea behind keep reconnecting is to provide support for the docker containers also for the testing purpose. If it keeps failing like that in the unavailability of istgt then it's an overhead for us to keep restarting the exporter container. I had discussion with @vishnuitta and @kmova regarding this, whether to keep it retrying to initiate the connection or  get out of the execution and we agreed on to let it be to retry indefinitely. This is the reason why i have written a go routine Retry() to make sure exporter container may never fail even if istgt container has failed.